src/toolkits/object_detection/object_detector.cpp

```diff
@@ -990,6 +1001,42 @@ std::unique_ptr<data_iterator> object_detector::create_iterator(
       iterator_params.annotation_position = data_iterator::annotation_position_enum::BOTTOM_LEFT;
   }
 
+  // Append empty annotations if no annotations are found
+  if (!iterator_params.data.contains_column(
+          iterator_params.annotations_column_name)) {
+    std::vector<flexible_type> dummy;
+    for (size_t i = 0; i < iterator_params.data.size(); i++) {
+      dummy.push_back(flex_list());
+    }
+    gl_sarray empty_ann(dummy, flex_type_enum::LIST);
+    iterator_params.data.add_column(empty_ann,
+                                    iterator_params.annotations_column_name);
+    iterator_params.has_annotations = false;
+  }
```

Comment:
Why is this needed? Shouldn't it be an error if the annotations column is missing?

Reply:
This is not relative to `evaluate()`, which I check the existence of annotation in the very beginning.
The change is for `predict()` with no annotations data.
Basically, the current OD data iterator assumes there "MUST" be an annotation column, or lots of function will crash. So what I did here is for the case for which the input data for `predict()` doesn't;t has ground truth, then I append a dummy one.
But yes I will break it down to several PRs.
Thanks!

Reply:
Ah, then let's just make the iterator robust to the `annotations_column_name` being empty, in which case there are no annotations. We shouldn't create an empty column just for this purpose

Reply:
Okay for sure :D
This might be a much bigger change tho.

Reply:
It shouldn't be. Make sure you understand how the data iterator works. We already support the predictions column name being empty.

Reply:
Alright I'll take a look!